HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs

```diff
@@ -57,23 +59,49 @@ private ICollection<Examination> GetPotentiallyAvailableAppointments(BasicAppoin
 
         public ICollection<Examination> SearchWithPriorities(AppointmentSearchWithPrioritiesDTO parameters)
         {
-            /*call BasicSearch(parameters.InitialParameters)
-             * if (emtpy)
-             *      relax parameters and call BasicSearch again
-             */
-            throw new NotImplementedException();
+            ICollection<Examination> freeAppointments = BasicSearch(parameters.InitialParameters);
+            if (freeAppointments.Count == 0)
+            {               
+                if (parameters.Priority == SearchPriority.Doctor)
+                    return RelaxDates(parameters);
+                else return RelaxDoctor(parameters);
+            }
+
+            return freeAppointments;
+        }
+
+        private ICollection<Examination> RelaxDoctor(AppointmentSearchWithPrioritiesDTO parameters)
+        {
+            ICollection<Doctor> allDoctors = _doctorRepository.GetAllDoctors();
```

Comment:
Unfortunately, GetDoctorsBySpecialty method or similar cannot be found in the repository, but it would make sense to add it to the repository instead of checking the speciality in the doctors class. I believe that method from doctors class could stay there for now, since maybe it will be needed in the future, but I will add GetDoctorsBySpecialty into the repository.

Reply:
Applied in 10f7608cb77efe6e2952e70b5775072ef31e3872.